### DISCUSSION TITLE
Regarding object detection from drone data using Geoai

### DISCUSSION THREAD

**Discussion by Arunaram-GIS** [opengeos/geoai#149] [Q&A] [answered]
Hi, i have used geoai to extract images and labels for automatic building detection using drone data and shapefile.the labels are not empty and has values 1-Buildings,0--ground.but when i start training the (valinfo:inf),i have made lots of changes still it is the same upto the end..but the other parameters are good.do help me with this pleases

**Comment by GISshady** [comment]
Intersted

**Comment by giswqs** [comment, marked as answer]
Follow the example here: https://geoai.gishub.org/examples/train_object_detection_model/

All you need is a drone imagery and a vector dataset representing buildings.

**Reply by Arunaram-GIS** [reply]
Thank u sir. Can the same can be used for tree detection from RGB drone imagery( to get tree count and shapefile of the same).my training samples must be point or polygon of the tree?. And also detect tree health in multispectral imagery

**Reply by Arunaram-GIS** [reply]
![predicted](https://github.com/user-attachments/assets/c0623562-d989-48dc-952b-0010232872c7)
![rgb](https://github.com/user-attachments/assets/1f513ad9-81b0-4f54-9ae0-98d50af52bb4)

These are the predicted buildings on the drone data but the training samples where taken from some other data,i could see that most of the buildings are merged....this was my parameter setting " geoai.object_detection(
    input_path="G:\\ML\\Pondicherry25\\ortho-pondy.tif",
    output_path="D:\\ARUNA\\ML\\14-4-25\\predicted_mask3.tif",
    model_path="D:\\ARUNA\\ML\\new_output4\\trained\\best_model.pth",
    window_size=512,
    overlap=256,
    confidence_threshold=0.4,
    batch_size=4,
    num_channels=3
)
geoai.train_MaskRCNN_model(
    images_dir="D://ARUNA//ML//new_output4//filtered_images",
    labels_dir="D://ARUNA//ML//new_output4//filtered_labels",
    output_dir="D://ARUNA//ML//new_output4//trained",
    num_channels=3,
    pretrained=True,
    batch_size=4,
    num_epochs=20,
    learning_rate=0.001,
   
" should i increase epochs..my training was good .as u showed in the video only..should i change any parameters,this drone data is 2cm resolution. It would be very helpful if you could help me with this
Once again thank u for ur time sir.Ur work is great..Keep inspiring

**Reply by Arunaram-GIS** [reply]
Can you please reply sir.should i increase epochs,or change any parameters?..this is  drone data from southern part of india.. might be the model is struggling bcoz the data are closely packed instead of less densly packed like other countries?
